src/Handlers/PaymentMethodHandler.php

```diff
@@ -55,4 +128,155 @@ public function finalize(
+                              "browserInfo": {
+                                "acceptHeader": "*/*",
+                                "colorDepth": 24,
+                                "language": "pt",
+                                "javaEnabled": false,
+                                "screenHeight": 2160,
+                                "screenWidth": 3840,
+                                "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64)",
+                                "timeZoneOffset": -120
+                              }
+                            }';
+
+        //TODO replace arg with $this->paymentStateDataService
+        //->getPaymentStateDataFromContextToken($salesChannelContext->getToken())
+        $stateData = $this->checkoutStateDataValidator->getValidatedAdditionalData(json_decode($stateDataJson, true));
+
+        //Build request
+        $request = array();
+        $request = $this->browserBuilder->buildBrowserData(
+            $_SERVER['HTTP_USER_AGENT'],
+            $_SERVER['HTTP_ACCEPT'],
+            $stateData['browserInfo']['screenWidth'],
+            $stateData['browserInfo']['screenHeight'],
+            $stateData['browserInfo']['colorDepth'],
+            $stateData['browserInfo']['timeZoneOffset'],
+            $stateData['browserInfo']['language'],
+            $stateData['browserInfo']['javaEnabled'],
+            $request
+        );
+        $request = $this->addressBuilder->buildDeliveryAddress(
+            $shippingStreetAddress['street'],
+            $shippingStreetAddress['houseNumber'],
+            $salesChannelContext->getShippingLocation()->getAddress()->getZipcode(),
+            $salesChannelContext->getShippingLocation()->getAddress()->getCity(),
+            $shippingState,
+            $salesChannelContext->getShippingLocation()->getAddress()->getCountry()->getIso(),
+            $request
+        );
+        $request = $this->addressBuilder->buildBillingAddress(
+            $billingStreetAddress['street'],
+            $billingStreetAddress['houseNumber'],
+            $salesChannelContext->getCustomer()->getActiveBillingAddress()->getZipcode(),
+            $salesChannelContext->getCustomer()->getActiveBillingAddress()->getCity(),
+            $billingState,
+            $salesChannelContext->getCustomer()->getActiveBillingAddress()->getCountry()->getIso(),
+            $request
+        );
+        $request = $this->paymentBuilder->buildPaymentData(
+            $salesChannelContext->getCurrency()->getIsoCode(),
+            $this->currency->sanitize(
+                $transaction->getOrder()->getPrice()->getTotalPrice(),
+                $salesChannelContext->getCurrency()->getIsoCode()
+            ),
+            $transaction->getOrder()->getOrderNumber(),
+            $this->configurationService->getMerchantAccount(),
+            $transaction->getReturnUrl(),
+            $request
+        );
+        $request = $this->customerBuilder->buildCustomerData(
```

Comment:
reminder

Reply:
Added a state.data check before assigning the values to the request